src/lib/ifaces/ethtool.rs

```diff
@@ -28,65 +31,180 @@ pub struct EthtoolPauseInfo {
     auto_negotiate: bool,
 }
 
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub struct EthtoolFeatureInfo {
+    #[serde(serialize_with = "ordered_map")]
+    pub fixed: HashMap<String, bool>,
+    #[serde(serialize_with = "ordered_map")]
+    pub changeable: HashMap<String, bool>,
+}
+
 #[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone, Default)]
 pub struct EthtoolInfo {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub pause: Option<EthtoolPauseInfo>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub features: Option<EthtoolFeatureInfo>,
+}
+
+fn ordered_map<S>(
+    value: &HashMap<String, bool>,
+    serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let ordered: BTreeMap<_, _> = value.iter().collect();
+    ordered.serialize(serializer)
```

Comment:
I like this ordered map for the options. Do you think it could be a good idea to extend it to bond/bridge/.. in the future? Thanks!

Reply:
We should. Let's track it at https://github.com/nispor/nispor/issues/130